airbyte-integrations/connectors/source-bigquery/source_bigquery/source.py

```diff
@@ -82,31 +113,105 @@ def check_connection(self, logger, config: Mapping[str, Any]) -> Tuple[bool, any
+                    except exceptions.HTTPError as error:
+                        self.logger.warning(f"Table had no inserts in the last seven days so no change history is available causing the error: {str(error.response.text)}")
+                        table_obj = IncrementalQueryResult(project_id, dataset_id, table_id, stream_name, where_clause, fallback_start=fallback_start, slice_range=slice_range, authenticator=self._auth)
+                if isinstance(table_obj, TableChangeHistory):
+                    concurrent_streams.append(table_obj.stream)
+                else:
+                    normal_streams.append(table_obj.stream)
+        for dataset in BigqueryDatasets(project_id=project_id, authenticator=self._auth).read_records(sync_mode=SyncMode.full_refresh):
             dataset_id = dataset.get("datasetReference")["datasetId"]
             # list and process each table under each base to generate the JSON Schema
-            for table_info in BigqueryTables(dataset_id=dataset_id, project_id=config["project_id"], authenticator=self._auth).read_records(sync_mode=SyncMode.full_refresh):
+            for table_info in BigqueryTables(dataset_id=dataset_id, project_id=project_id, authenticator=self._auth).read_records(sync_mode=SyncMode.full_refresh):
                 table_id = table_info.get("tableReference")["tableId"]
                 if sync_method == "Standard":
-                    table_obj = IncrementalQueryResult(config["project_id"], dataset_id, table_id, authenticator=self._auth)
+                    table_obj = IncrementalQueryResult(project_id, dataset_id, table_id, fallback_start=fallback_start, slice_range=slice_range, authenticator=self._auth)
                 else:
                     try:
-                        table_obj = TableChangeHistory(config["project_id"], dataset_id, table_id, authenticator=self._auth)
+                        table_obj = TableChangeHistory(project_id, dataset_id, table_id, fallback_start=change_history_start, slice_range=slice_range, authenticator=self._auth)
                         next(table_obj.read_records(sync_mode=SyncMode.full_refresh))
-                    except Exception as e:
-                        self.logger.warn(str(e))
-                        table_obj = IncrementalQueryResult(config["project_id"], dataset_id, table_id, authenticator=self._auth)
-                streams_catalog.append(table_obj.stream)
-        for stream in streams_catalog:
-            yield stream
+                    except exceptions.HTTPError as error:
+                        self.logger.warning(f"Table had no inserts in the last seven days so no change history is available causing the error: {str(error.response.text)}")
+                        table_obj = IncrementalQueryResult(project_id, dataset_id, table_id, fallback_start=fallback_start, slice_range=slice_range, authenticator=self._auth)
+                
+                if isinstance(table_obj, TableChangeHistory):
```

Comment:
I did not make it concurrent because of the concurrent `DefaultStream` forces the streams to have `source_defined_cursor=true` (as I discussed with you in slack) and this stream should have a user defined cursor. 
I have a work around in mind; I am thinking of only making this stream concurrent after the catalog is set which means only during the read, what do you think of this solution?

Reply:
> I did not make it concurrent because the concurrent DefaultStream forces the streams to have source_defined_cursor=true (as I discussed with you in slack) and this stream should have a user defined cursor.

I think we can still make it part of this method. We just need to have the source get the catalog during `__init__`. What do you think about that? If the catalog is not defined, no need to have these concurrent as it means this is a discover or check call

Reply:
@maxi297 
I made a workaround in order to make `IncrementalQueryResult ` also concurrent, here's the commit https://github.com/artefactory/airbyte/pull/30/commits/b121f38df041f475fd3b672b5292430d7ed4f966

Note: the chosen cursor field should always be of type `datetime` object (since ConcurrentCursor expects a datetime or timestamp object), I did not enforce this on the UI but if the user choses a field that's not of type `datetime` as cursor and error is raised